File: src/main/java/io/curiousoft/ijudi/ordermanagement/model/Device.java
package io.curiousoft.ijudi.ordermanagement.model;

import org.springframework.data.mongodb.core.index.Indexed;

import javax.validation.constraints.NotBlank;
import java.util.UUID;

public class Device extends BaseModel {

    @NotBlank(message = "device token required")
    @Indexed(unique = true)
    private String token;
    private String userId;

    public Device(@NotBlank(message = "device token required") String token) {
        super(UUID.randomUUID().toString());
        this.token = token;
        this.userId = userId;
    }

    public Device() {
        super(null);
        this.token = token;
        this.userId = userId;
    }

    public String getToken() {
        return token;
    }

    public void setToken(String token) {
        this.token = token;
    }

    public String getUserId() {
        return userId;
    }

    public void setUserId(String userId) {
        this.userId = userId;
    }
}

File: izinga-commons/src/main/kotlin/io/curiousoft/izinga/commons/model/Device.kt
package io.curiousoft.izinga.commons.model

import org.springframework.data.mongodb.core.index.Indexed
import java.util.*
import javax.validation.constraints.NotBlank

class Device : BaseModel {
    @Indexed(unique = true)
    var token: @NotBlank(message = "device token required") String? = null
    var userId: String? = null

    constructor(token: @NotBlank(message = "device token required") String?) : super(UUID.randomUUID().toString()) {
        this.token = token
    }

    constructor() : super(null)
}